File: typing/src/vo/KvList.ts
import {Kv} from './Kv.ts';

export class KvList {
    list: Kv[];

    constructor(list: Kv[]) {
        this.list = list;
    }

    static fromJson(json: any): KvList {
        return new KvList(
            (json.list ?? []).map((e: any) => Kv.fromJson(e))
        );
    }

    convertMap(): Map<string, string> {
        const map = new Map<string, string>();

        for (const item of this.list) {
            map.set(item.k, item.v);
        }

        return map;
    }
}

File: lib/logic/game_server/vo/kv_list.dart
import 'kv.dart';

class KvList {
  final List<Kv> list;

  KvList(this.list);

  Map<String, dynamic> toJson() {
    return {
      'list': list.map((e) => e.toJson()).toList(),
    };
  }

  factory KvList.fromJson(Map<String, dynamic> json) {
    return KvList(
      (json['list'] as List<dynamic>).map((e) => Kv.fromJson(e as Map<String, dynamic>)).toList(),
    );
  }
}
